src/backend/mapServer.js

```diff
@@ -0,0 +1,69 @@
+const { promisify } = require("util");
+const getPort = require("get-port");
+const createServer = require("@mapeo/map-server").default;
+const log = require("debug")("map-server");
+
+const AsyncService = require("./upgrade-manager/async-service");
+
+// TODO: Does it make sense for this to extend AsyncService?
+class MapServer extends AsyncService {
```

Comment:
using the async-service seems to make sense here, but open to other thoughts. didn't want to bother with moving files around so it's currently still under the upgrade manager directory, even though it's more general

Reply:
Yes that makes sense. I'm thinking of changing that API for this slightly, to make it into its own state machine, so that rather than extending the AsyncService class you do something like:

```js
  constructor() {
    #service = new AsyncStateMachine({ start, stop })
  }
```

We can do that in a follow-up PR though, I'll extract the start/stop service management from the project invite PR.